File: sdkwork-backend-sdk-swift/Sources/API/AppApi.swift
import Foundation

public class AppApi {
    private let client: HttpClient
    
    public init(client: HttpClient) {
        self.client = client
    }

    /// Update application
    public func update(body: PlusAppForm) async throws -> PlusApiResultPlusAppVO? {
        let response = try await client.put(ApiPaths.backendPath("/app"), body: body)
        return response as? PlusApiResultPlusAppVO
    }

    /// Create application
    public func create(body: PlusAppForm) async throws -> PlusApiResultPlusAppVO? {
        let response = try await client.post(ApiPaths.backendPath("/app"), body: body)
        return response as? PlusApiResultPlusAppVO
    }

    /// Get applications by page
    public func listByPage(body: QueryListForm? = nil, params: [String: Any]? = nil) async throws -> PlusApiResultPagePlusAppVO? {
        let response = try await client.post(ApiPaths.backendPath("/app/list"), body: body, params: params)
        return response as? PlusApiResultPagePlusAppVO
    }

    /// Get all applications
    public func listAllEntities(body: QueryListForm? = nil) async throws -> PlusApiResultListPlusAppVO? {
        let response = try await client.post(ApiPaths.backendPath("/app/list/all"), body: body)
        return response as? PlusApiResultListPlusAppVO
    }

    /// Get app SDK config
    public func getSdkConfig(body: CreateJsapiSignatureForm) async throws -> PlusApiResultAppSdkConfigVO? {
        let response = try await client.post(ApiPaths.backendPath("/app/get_sdk_config"), body: body)
        return response as? PlusApiResultAppSdkConfigVO
    }

    /// Get application by ID
    public func getById(id: String) async throws -> PlusApiResultPlusAppVO? {
        let response = try await client.get(ApiPaths.backendPath("/app/\(id)"))
        return response as? PlusApiResultPlusAppVO
    }

    /// Delete application
    public func delete(id: String) async throws -> PlusApiResultBoolean? {
        let response = try await client.delete(ApiPaths.backendPath("/app/\(id)"))
        return response as? PlusApiResultBoolean
    }

    /// Get app info
    public func getAppInfo() async throws -> PlusApiResultAppInfoVO? {
        let response = try await client.get(ApiPaths.backendPath("/app/info"))
        return response as? PlusApiResultAppInfoVO
    }
}

File: sdkwork-backend-sdk-csharp/Api/AppApi.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using Backend.Http;
using Backend.Models;

namespace Backend.Api
{
    public class AppApi
    {
        private readonly HttpClient _client;

        public AppApi(HttpClient client)
        {
            _client = client;
        }

        /// <summary>
        /// Update application
        /// </summary>
        public async Task<PlusApiResultPlusAppVO?> UpdateAsync(PlusAppForm body)
        {
            return await _client.PutAsync<PlusApiResultPlusAppVO>(ApiPaths.BackendPath("/app"), body);
        }

        /// <summary>
        /// Create application
        /// </summary>
        public async Task<PlusApiResultPlusAppVO?> CreateAsync(PlusAppForm body)
        {
            return await _client.PostAsync<PlusApiResultPlusAppVO>(ApiPaths.BackendPath("/app"), body);
        }

        /// <summary>
        /// Get applications by page
        /// </summary>
        public async Task<PlusApiResultPagePlusAppVO?> ListByPageAsync(QueryListForm? body = null, Dictionary<string, object>? query = null)
        {
            return await _client.PostAsync<PlusApiResultPagePlusAppVO>(ApiPaths.BackendPath("/app/list"), body, query);
        }

        /// <summary>
        /// Get all applications
        /// </summary>
        public async Task<PlusApiResultListPlusAppVO?> ListAllEntitiesAsync(QueryListForm? body = null)
        {
            return await _client.PostAsync<PlusApiResultListPlusAppVO>(ApiPaths.BackendPath("/app/list/all"), body);
        }

        /// <summary>
        /// Get app SDK config
        /// </summary>
        public async Task<PlusApiResultAppSdkConfigVO?> GetSdkConfigAsync(CreateJsapiSignatureForm body)
        {
            return await _client.PostAsync<PlusApiResultAppSdkConfigVO>(ApiPaths.BackendPath("/app/get_sdk_config"), body);
        }

        /// <summary>
        /// Get application by ID
        /// </summary>
        public async Task<PlusApiResultPlusAppVO?> GetByIdAsync(string id)
        {
            return await _client.GetAsync<PlusApiResultPlusAppVO>(ApiPaths.BackendPath($"/app/{id}"));
        }

        /// <summary>
        /// Delete application
        /// </summary>
        public async Task<PlusApiResultBoolean?> DeleteAsync(string id)
        {
            return await _client.DeleteAsync<PlusApiResultBoolean>(ApiPaths.BackendPath($"/app/{id}"));
        }

        /// <summary>
        /// Get app info
        /// </summary>
        public async Task<PlusApiResultAppInfoVO?> GetAppInfoAsync()
        {
            return await _client.GetAsync<PlusApiResultAppInfoVO>(ApiPaths.BackendPath("/app/info"));
        }
    }
}
